### ISSUE TITLE
Remove isarray dependency 

### ISSUE THREAD

**Issue by jimmywarting** [nodejs/readable-stream#212]
think we can replace it with `Array.isArray`

**Comment by calvinmetcalf** [comment]
Not on all targeted platforms I don't believe

**Comment by mcollina** [comment]
Is `Array.isArray` present everywhere on all our target platforms?

**Comment by jimmywarting** [comment]
Wich platform dö we target?

**Comment by calvinmetcalf** [comment]
actually now that I look at it, isArray seems to be available on all the platforms, we can test

**Comment by dcousineau** [comment]
The `isarray` package in its entirety is:

```javascript
var toString = {}.toString;

module.exports = Array.isArray || function (arr) {
  return toString.call(arr) == '[object Array]';
};
```

And can more simply be inlined, not required.

**Comment by Croydon** [comment]
Before I have opened #318 I should be searched first. Shame on me :) 

So what's the plan here? You discussed in the pull request which was going to remove the isArray dependency, then you decided to make it inline https://github.com/nodejs/readable-stream/commit/69d13e3747f7c3596a6cbaf266f43f6d8e5fcb7d#diff-279988bef5e6fbf7ed82cae28ecc2d59 just to revert this again in the next commit https://github.com/nodejs/readable-stream/commit/ea4eabaa30929f685ec1c3955587ec4513b31ae8#diff-279988bef5e6fbf7ed82cae28ecc2d59

So...what is the plan?

**Comment by jimmywarting** [comment]
Bumping my thread 2 years later after still seeing it being used in my dependency graph from karma...

Can we bump the major version soon and remove the dependency and perhaps the inlined version also?

looks pretty green to me:
![skarmavbild 2018-04-27 kl 20 03 32](https://user-images.githubusercontent.com/1148376/39377645-33ec66d0-4a56-11e8-9ed7-1d1e451fa63b.png)

**Comment by mcollina** [comment]
Yes, we will do that in the coming weeks. We will cut a new major release from Node 10 as soon as possible (but it will take a while).
